### PR TITLE
Prevent Folders from being previewed

### LINKED ISSUE #2901: Able to apply Object styles / Use Conditional Styling... for non editable objects
**Steps to reproduce:**
1. Add any non editable object under 'My Items' (eg: Notebook)
1. Check Notebook inspector - styles
1. Make sure you can not add/edit styles.
1. Now add any editable object under 'My Items' (eg: Overlay Plot)
1. While in edit mode, click on 'My Items', this should open Preview window with 'My items'
1. In the preview overlay, click on a non-editable object (eg: Notebook) (nothing happens on Preview but center layout now has Notebook)
1. Press 'Done' to close Preview.
1. Now in you are in edit mode for non-editable object (eg: Notebook).
1. You can add styles on it.

**Alternate Way**
1. Take snapshot of 'My Items', so its in 'Notebook Snapshots'
1. While in edit mode,  click 'Preview' from context menu of that snapshot in Notebook Snapshots.

Screen Shot 2020-04-09 at 12 25 48 PM

Screen Shot 2020-04-09 at 11 42 48 AM

Screen Shot 2020-04-09 at 11 38 09 AM

### PR DESCRIPTION
**Pull request by charlesh88**
Fixes #2901. Mods the PreviewAction to prevent Folders from being previewed.

### Author Checklist
1. Changes address original issue? Y
1. Unit tests included and/or updated with changes? N/A
1. Command line build passes? Y
1. Changes have been smoke-tested? Y

### CODE REVIEW

**Comment by nikhilmandlik** [comment]
Able to reproduce using another method:
Take notebook snapshot of 'My Items' then click 'Preview' from embed menu and navigate to non editable object.

**Comment by charlesh88** [comment]
> Able to reproduce using another method:
> Take notebook snapshot of 'My Items' then click 'Preview' from embed menu and navigate to non editable object.

So why isn't the link in the embed using PreviewAction?

File: src/ui/preview/PreviewAction.js [hunk 1]
```diff
@@ -63,12 +63,16 @@ export default class PreviewAction {
         });
     }
     appliesTo(objectPath) {
-        return !this._isNavigatedObject(objectPath)
+        return !this._isNavigatedObject(objectPath) && !this._preventPreview(objectPath);
```
Comment (nikhilmandlik) [inline src/ui/preview/PreviewAction.js:66]: Nice, thanks @charlesh88

**Comment by akhenry** [comment]
## Reviewer Checklist

* Changes appear to address issue? Y
* Appropriate unit tests included? N/A
* Code style and in-line documentation are appropriate? Y
* Commit messages meet standards? Y

**Comment by charlesh88** [comment]
Testing notes:
1. Create a folder under "My Items" if one does not already exist.
1. Context-click the folder. Verify that there is no "Preview" option in the menu.

Known issues:
1. "My Items" currently allows Preview.

**Pull request merged**